launcher/ui/instanceview/VisualGroup.cpp

```diff
@@ -112,183 +101,105 @@ QPair<int, int> VisualGroup::positionOf(const QModelIndex &index) const
+    int textOffsetLeft = arrowOffsetLeft * 2;
+    int arrowSize = 6;
 
-    //BEGIN: right vertical line
+    // BEGIN: arrow
     {
-        QPoint start(optRect.topRight());
-        start.ry() += 3;
-        QPoint verticalGradBottom(optRect.topRight());
-        verticalGradBottom.ry() += fontMetrics.height() + 5;
-        QLinearGradient gradient(start, verticalGradBottom);
-        gradient.setColorAt(0, outlineColor);
-        gradient.setColorAt(1, Qt::transparent);
-        painter->fillRect(QRect(start, QSize(1, fontMetrics.height() + 5)), gradient);
-    }
-    //END: right vertical line
-
-    //BEGIN: checkboxy thing
-    {
-        painter->save();
         painter->setRenderHint(QPainter::Antialiasing, false);
-        painter->setFont(font);
-        QColor penColor(option.palette.text().color());
-        penColor.setAlphaF(0.6);
-        painter->setPen(penColor);
-        QRect iconSubRect(option.rect);
-        iconSubRect.setTop(iconSubRect.top() + 7);
-        iconSubRect.setLeft(iconSubRect.left() + 7);
-
-        int sizing = fontMetrics.height();
-        int even = ( (sizing - 1) % 2 );
-
-        iconSubRect.setHeight(sizing - even);
-        iconSubRect.setWidth(sizing - even);
-        painter->drawRect(iconSubRect);
-
+        painter->save();
 
-        /*
-        if(collapsed)
-            painter->drawText(iconSubRect, Qt::AlignHCenter | Qt::AlignVCenter, "+");
-        else
-            painter->drawText(iconSubRect, Qt::AlignHCenter | Qt::AlignVCenter, "-");
-        */
-        painter->setBrush(option.palette.text());
-        painter->fillRect(iconSubRect.x(), iconSubRect.y() + iconSubRect.height() / 2,
-                          iconSubRect.width(), 2, penColor);
-        if (collapsed)
-        {
-            painter->fillRect(iconSubRect.x() + iconSubRect.width() / 2, iconSubRect.y(), 2,
-                              iconSubRect.height(), penColor);
+        QPolygon polygon;
+        if (collapsed) {
+            polygon << QPoint(arrowOffsetLeft - arrowSize / 2, centerHeight - arrowSize)
+                    << QPoint(arrowOffsetLeft + arrowSize / 2, centerHeight)
+                    << QPoint(arrowOffsetLeft - arrowSize / 2, centerHeight + arrowSize);
+            painter->drawPolyline(polygon);
+        } else {
+            polygon << QPoint(arrowOffsetLeft - arrowSize, centerHeight - arrowSize / 2)
+                    << QPoint(arrowOffsetLeft, centerHeight + arrowSize / 2)
+                    << QPoint(arrowOffsetLeft + arrowSize, centerHeight - arrowSize / 2);
+            painter->drawPolyline(polygon);
         }
-
         painter->restore();
     }
-    //END: checkboxy thing
+    // END: arrow
 
-    //BEGIN: text
+    // BEGIN: text
     {
-        QRect textRect(option.rect);
-        textRect.setTop(textRect.top() + 7);
-        textRect.setLeft(textRect.left() + 7 + fontMetrics.height() + 7);
+        painter->setRenderHint(QPainter::Antialiasing);
```

Comment:
If you move `setRenderHint` after `save` this probably isn't needed

Reply:
after reading the docs for save and restore I think actually this isn't needed at all.